### PR TITLE
Fix all shellcheck issues and add it to local lint target

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, this

1. Lints all shell files
2. Fixes all existing issues
3. Adds shellcheck to `make lint` so it can be run locally easily

/assign @cardil

### CODE REVIEW

File: test/lib.bash [hunk 1]
```diff
@@ -389,13 +389,14 @@ function create_htpasswd_users {
   if kubectl get secret htpass-secret -n openshift-config -o jsonpath='{.data.htpasswd}' 2>/dev/null | base64 -d > users.htpasswd; then
     logger.info 'Secret htpass-secret already existed, updating it.'
     # Add a newline to the end of the file if not already present (htpasswd will butcher it otherwise).
+    # shellcheck disable=SC1003
     sed -i -e '$a\' users.htpasswd
```
Comment (cardil) [inline test/lib.bash:393]: I saw that. I don't know it that a problem or not. It looks weird, like a bug.

Warren wrote that. @warrenvw, maybe you remember what that suppose to achieve?

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -83,10 +83,7 @@ jobs:
         env:
           REVIEWDOG_GITHUB_API_TOKEN: ${{ github.token }}
         run: |
-          find . -type f \
-            -path './hack/*.*sh' \
-            -o -path './test/*.*.sh' \
-            -not -path '*vendor*' | xargs -r shellcheck --format=checkstyle \
+          find . -type f -path './**/*.*sh' -not -path '*vendor*' | xargs -r shellcheck --format=checkstyle \
```
Comment (cardil) [inline .github/workflows/validate.yaml:86]: Why this reformatting here?

File: hack/lib/catalogsource.bash [hunk 3]
```diff
@@ -176,6 +176,7 @@ function add_user {
   logger.info "Creating user $name:***"
   if kubectl get secret htpass-secret -n openshift-config -o jsonpath='{.data.htpasswd}' 2>/dev/null | base64 -d > users.htpasswd; then
     logger.debug 'Secret htpass-secret already existed, updating it.'
+    # shellcheck disable=SC1003
```
Comment (cardil) [inline hack/lib/catalogsource.bash:179]: Same as the other comment.

**Review by cardil — COMMENTED** [review]
Cool :heart:!

Some doubts/questions...

File: test/lib.bash (hunk 1 above)
Comment (markusthoemmes) [reply test/lib.bash:393]: Well it achieves what the comment states. A quick google told me that this is indeed the notation for that in `sed` (I doublechecked this too)

File: .github/workflows/validate.yaml (hunk 2 above)
Comment (markusthoemmes) [reply .github/workflows/validate.yaml:86]: easier to read (imo anyway)

File: test/lib.bash (hunk 1 above)
Comment (cardil) [reply test/lib.bash:393]: Maybe better would be to rewrite that to simpler and POSIX compliant alternative: https://unix.stackexchange.com/a/263965/145501

File: .github/workflows/validate.yaml (hunk 2 above)
Comment (cardil) [reply .github/workflows/validate.yaml:86]: For me the opposite is true. The original is easier to read and understand. :rofl:

File: test/lib.bash (hunk 1 above)
Comment (markusthoemmes) [reply test/lib.bash:393]: Indeed, changing!

File: .github/workflows/validate.yaml (hunk 2 above)
Comment (markusthoemmes) [reply .github/workflows/validate.yaml]: Changed :)

**Review by cardil — APPROVED** [review]
/lgtm

**Pull request merged**